### PR TITLE
docreq: Add request type and budget code fields

### LINKED ISSUE #682: Document request: add a new field loan/buy
When creating a new document request, the user should be able to choose if he would like to buy or loan the book, (s)he will have to fill the Document Request form. The form needs to be enhanced with the new required field `Request type` and values `Loan` and `Buy.`

Budget code should also be added.

Frontend: inveniosoftware/react-invenio-app-ils/pull/101

### PR DESCRIPTION
**Pull request by jrcastro2**
* Add request type field to JSON schema, mappings
  and JSON loaders for Document Requests

* Add budget code field to JSON schema, mappings
  and JSON loaders for Document Requests

* closes #682

### CODE REVIEW

File: invenio_app_ils/document_requests/mappings/v6/document_requests/document_request-v1.0.0.json [hunk 1]
```diff
@@ -16,6 +16,9 @@
         "authors": {
           "type": "text"
         },
+        "budget_code": {
```
Comment (kpsherva) [inline invenio_app_ils/document_requests/mappings/v6/document_requests/document_request-v1.0.0.json]: let's have a chat about this field name

File: invenio_app_ils/document_requests/loaders/jsonschemas/document_request.py [hunk 2]
```diff
@@ -56,9 +56,12 @@ class Meta:
     note = SanitizedHTML()
     page = SanitizedHTML()
     patron_pid = fields.Str(required=True, validate=validate_patron)
+    payment_info = fields.Str()
```
Comment (topless) [inline invenio_app_ils/document_requests/loaders/jsonschemas/document_request.py:60]: Food for thought, I am not sure if I would go for the explicit `payment_*` fields or if I would create a complex field `payment` with `info` and `method` as properties, very similar to `price`, @kprzerwa, @ntarocco

File: invenio_app_ils/document_requests/schemas/document_requests/document_request-v1.0.0.json [hunk 3]
```diff
@@ -8,6 +8,7 @@
       "title": "Authors of the requested document",
       "type": "string"
     },
+
```
Comment (topless) [inline invenio_app_ils/document_requests/schemas/document_requests/document_request-v1.0.0.json]: nit: unecessary

File: run-style-checks.sh [hunk 4]
```diff
@@ -7,6 +7,6 @@
 # under the terms of the MIT License; see LICENSE file for more details.
 
 pydocstyle invenio_app_ils tests docs && \
-isort -rc -c -df && \
+isort --check-only --diff && \
```
Comment (topless) [inline run-style-checks.sh]: Missed the recursive part
```suggestion
isort --recursive --check-only --diff && \
```

File: invenio_app_ils/document_requests/loaders/jsonschemas/document_request.py (hunk 2 above)
Comment (kpsherva) [reply invenio_app_ils/document_requests/loaders/jsonschemas/document_request.py:60]: yes maybe consult this as a team

File: run-style-checks.sh (hunk 4 above)
Comment (jrcastro2) [reply run-style-checks.sh]: In isort v5 `--recursive` option has been removed. Directories passed in are now automatically sorted recursive.

**Review by topless — APPROVED** [review]

**Review by ntarocco — COMMENTED** [review]
You are going all to be mad at me, I know already! :)
But I would have created 2 radio buttons (interested in buying, interested in loaning) and a budget code field.
I am not sure a vocabulary is needed since I doubt there might me more possibilities, and the budget code in the end we added to ILL and ACQ...

**Pull request merged**